File: Retos/Reto #3 - EL GENERADOR DE CONTRASEÑAS [Media]/typescript/KevinED11.ts
  /*
  * Escribe un programa que sea capaz de generar contraseñas de forma aleatoria.
  * Podrás configurar generar contraseñas con los siguientes parámetros:
  * - Longitud: Entre 8 y 16.
  * - Con o sin letras mayúsculas.
  * - Con o sin números.
  * - Con o sin símbolos.
  * (Pudiendo combinar todos estos parámetros entre ellos)
  */




  const passwordGenerator = (longitud: number = 10, uppercase: boolean = true, 
      numeros: boolean = true, simbols: boolean= true): string => {

    if (longitud > 8 && longitud <= 16) {

      let password: string = ""
      let characters = "abcdefghijklmnopqrstuvwxyz"
      
      characters += (numeros ? "0123456789" : "") 
      + (simbols ? "!@#$%^&*()_+-={}[]|\\:;\\\">,.?/": "") 
      + (uppercase ? "ABCDEFGHIJKLMNOPQRSTUVWXYZ" : "")
    
      for (let i = 0; i < longitud; i++ ) {
        password += characters.charAt(Math.floor(Math.random() * characters.length))
        

    } 
      return password

    } 

    return "Coloca entre 8 y 16 caracteres de longitud"

  }

  console.log(passwordGenerator(15))

File: Retos/Reto #3 - EL GENERADOR DE CONTRASEÑAS [Media]/python/KevinED11.py
from string import ascii_lowercase, ascii_uppercase, punctuation, digits
from random import choice

# por defecto genera una contraseña con una longitud de 10 caracteres y con todos los parámetros


def pass_gen(length: int = 10, all_params: bool = True, upper: bool = False, lower: bool = False,  numbers: bool = False, simbols: bool = False) -> str:

    characters = ""

    char_all = ascii_lowercase + ascii_uppercase + punctuation + digits

    if length not in list(range(8, 17)):
        return 'porfavor vuelve a intentarlo, debes colocar un rango entre 8 y 16 caracteres.'

    if all_params:
        characters += char_all
    elif upper:
        characters += ascii_uppercase
    elif numbers:
        characters += digits
    elif simbols:
        characters += punctuation

    contraseña = "".join(choice(characters) for i in range(length))

    return f'su nueva contraseña generada es: {contraseña}'


print(pass_gen())
